lib/PuppeteerSharp/JSCoverage.cs

```diff
@@ -63,7 +65,91 @@ public JSCoverage(Session client)
            );
             _client.MessageReceived -= client_MessageReceived;
 
-            // TODO: return coverage;            
+            var coverage = new List<CoverageEntry>();
+            foreach (var entry in profileResponseTask.Result.Result)
+            {
+                _scriptURLs.TryGetValue(entry.ScriptId, out var url);
+                _scriptSources.TryGetValue(entry.ScriptId, out var text);
+
+                if (text == null || url == null)
+                {
+                    continue;
+                }
+
+                var flattenRanges = new List<ProfilerTakePreciseCoverageResponseRange>();
```

Comment:
we could use a `SelectMany` here instead of `foreach` & `AddRange`
```cs
var flattenRanges = entry.Functions
    .SelectMany(func => func.Ranges)
    .ToList();
```